src/main/java/com/flightstats/hub/channel/ChannelResource.java

```diff
@@ -71,53 +70,53 @@
     @GET
     @Produces(MediaType.APPLICATION_JSON)
     public Response getChannelMetadata(@PathParam("channel") String channelName,
-                                       @QueryParam("cached") @DefaultValue("true") boolean cached) {
+                                       @QueryParam("cached") @DefaultValue("true") boolean cached) throws WebApplicationException {
         logger.debug("get channel {}", channelName);
 
-        Optional<ChannelConfig> optionalChannelConfig = channelService.getChannelConfig(channelName, cached);
-        if (!optionalChannelConfig.isPresent()) {
-            logger.info("unable to get channel " + channelName);
-            throw new WebApplicationException(Response.Status.NOT_FOUND);
-        }
+        ChannelConfig channelConfig = channelService.getChannelConfig(channelName, cached)
+                .orElseThrow(() -> {
+                    logger.info("unable to get channel " + channelName);
+                    throw new WebApplicationException(Response.Status.NOT_FOUND);
+                });
 
-        ObjectNode output = buildChannelConfigResponse(optionalChannelConfig.get(), uriInfo, channelName);
+        ObjectNode output = buildChannelConfigResponse(channelConfig, uriInfo, channelName);
         return Response.ok(output).build();
     }
 
     @PUT
     @Produces(MediaType.APPLICATION_JSON)
     @Consumes(MediaType.APPLICATION_JSON)
-    public Response createChannel(@PathParam("channel") String channelName, String json) throws Exception {
+    public Response createChannel(@PathParam("channel") String channelName, String json) {
         logger.debug("put channel {} {}", channelName, json);
-        Optional<ChannelConfig> optionalChannelConfig = channelService.getChannelConfig(channelName, false);
-        ChannelConfig channelConfig = ChannelConfig.createFromJsonWithName(json, channelName);
-        if (optionalChannelConfig.isPresent()) {
-            ChannelConfig oldConfig = optionalChannelConfig.get();
-            logger.info("using old channel {} {}", oldConfig, oldConfig.getCreationDate().getTime());
-            channelConfig = ChannelConfig.updateFromJson(oldConfig, StringUtils.defaultIfBlank(json, "{}"));
-        }
-        logger.info("creating channel {} {}", channelConfig, channelConfig.getCreationDate().getTime());
-        channelConfig = channelService.updateChannel(
-                channelConfig,
-                optionalChannelConfig.orElse(null),
+        ChannelConfig newChannelConfig = ChannelConfig.createFromJsonWithName(json, channelName);
+        
+        ChannelConfig oldChannelConfig = channelService
+                .getChannelConfig(channelName, false)
+                .orElse(newChannelConfig);
+
+
+        logger.info("creating channel {} {}", newChannelConfig, newChannelConfig.getCreationDate().getTime());
+        newChannelConfig = channelService.updateChannel(
```

Comment:
I'm not a fan of using `newChannelConfig` here because of mutability (even though it's scoped to the method). I'd create a new `channelConfig` or `updatedChannelConfig` or something instead.